app/types/bool.go

```diff
@@ -24,54 +17,30 @@ type (
 	OptNullBool struct{ Bool }
 )
 
-// NewBool creates a Bool which is not-null and set with the given value
+// NewBool creates a Bool which is not-null and set with the given Value
 func NewBool(s bool) *Bool {
-	n := &Bool{}
-	n.Value = s
-	n.Set = true
-	n.Null = false
+	n := &Bool{Data{Value: s, Set: true, Null: false}}
 	return n
 }
 
 // UnmarshalJSON parse JSON data to the type
 func (s *Bool) UnmarshalJSON(data []byte) (err error) {
-	s.Set = true // If this method was called, the value was set.
-	s.Null = string(data) == jsonNull
-	var temp bool
-	err = json.Unmarshal(data, &temp)
-	if err == nil {
-		s.Value = temp
-	}
-	return
-}
-
-// AllAttributes unwrap the wrapped value and its attributes
-func (s Bool) AllAttributes() (value interface{}, isNull, isSet bool) {
-	return s.Value, s.Null, s.Set
+	return unmarshalJSON(data, &s.Set, &s.Null, &s.Value, reflect.TypeOf(true))
```

Comment:
Unfortunately I can't because Int64 calls unmarshalJSON in another way:
```
	var temp string
	err = unmarshalJSON(data, &i.Set, &i.Null, &temp, reflect.TypeOf(""))
```